odoo/addons/base/models/res_partner_bank.py

```diff
@@ -0,0 +1,208 @@
+            if label == 'jp' and (len(bank.clearing_number) != 3 or not bank.clearing_number.isdecimal()):
+                raise ValidationError(self.env._("Zengin Branch Code must consist of 3 digits."))
+            if label == 'sa' and (len(bank.clearing_number) != 4 or not bank.clearing_number.isalpha() or not bank.clearing_number.isupper()):
+                raise ValidationError(self.env._("Bank SARIE ID must consist of 4 upper case English letter."))
+            if label == 'us' and not re.match(r'^\d{1,9}$', bank.clearing_number):
+                raise ValidationError(self.env._("Routing Number should only contain numbers (maximum 9 digits)."))
+
+    def _get_clearing_number(self, clearing_number_label):
+        if not self:
+            return None
+        self.ensure_one()
+        return self.clearing_number_label == clearing_number_label and self.clearing_number
+
+    @api.model
+    def _get_clearing_number_label_by_bank_country(self):
+        """
+        :returns: the selection value of clearing_number_label corresponding to the bank country if any otherwise returns 'all'
+        """
+        self.ensure_one()
+        if (
+            (country_code := (self.country_code or '').lower())
+            and country_code in self._fields['clearing_number_label']._selection
+        ):
+            return country_code
+        else:
+            return 'all'
+
+    @api.depends('bank_country_id')
+    def _compute_clearing_number_label(self):
+        for bank in self:
+            bank.clearing_number_label = bank._get_clearing_number_label_by_bank_country()
+
+    @api.depends('bank_country_id')
+    def _compute_show_clearing_number(self):
+        sepa_countries = self.env.ref('base.sepa_zone').country_ids
+        for bank in self:
+            bank.show_clearing_number = bank._get_clearing_number_label_by_bank_country() != 'all' or bank.bank_country_id not in sepa_countries
+
+    @api.depends('acc_number')
+    def _compute_sanitized_acc_number(self):
+        for bank in self:
+            bank.sanitized_acc_number = sanitize_account_number(bank.acc_number)
+
+    @api.depends('partner_id.country_id', 'company_id.country_id')
+    def _compute_bank_country_id(self):
+        for bank in self:
+            if not bank.bank_country_id:
+                bank.bank_country_id = bank.partner_id.country_id or bank.company_id.country_id or self.env.company.country_id
+
+    def open_linked_partner_id(self):
+        return self.partner_id._get_records_action()
+
+    def _search_acc_number(self, operator, value):
+        if operator in ('in', 'not in'):
+            value = [sanitize_account_number(i) for i in value]
+        else:
+            value = sanitize_account_number(value)
+        return [('sanitized_acc_number', operator, value)]
+
+    @api.depends('acc_number')
+    def _compute_acc_type(self):
+        for bank in self:
+            bank.acc_type = bank.retrieve_acc_type(bank.acc_number)
```

Comment:
Just override this function in other modules and remove `retrieve_acc_type`.
```suggestion
    @api.depends('acc_number')
    def _compute_acc_type(self):
        self.acc_type = 'bank'
```